src/controllers/NotificationController.ts

```diff
@@ -0,0 +1,59 @@
+/**
+ * Tests for Notification Controllers
+ * Send HTTP requests to Notification Controllers and check mocked responses
+ */
+
+import * as express from "express";
+import { NotificationAPI } from "../api/services/NotificationAPI";
+import { NotificationSubscriptionResponse } from "../api/types/NotificationSubscriptionResponse";
+import { ControllerError } from "../enums/ControllerError";
+import { HttpErrorStatusCode } from "../enums/HttpErrorStatusCode";
+import { NotificationSubscriptionRequestType } from "../enums/NotificationSubscriptionType";
+import { AppResponseConverter } from "../utils/AppResponseConverter";
+import { RestfulUtils } from "../utils/RestfulUtils";
+
+// Notification Controller
+export class NotificationController {
+  // Update user subscription to Notification Service
+  public static updateSubscription(
+    req: express.Request,
+    res: express.Response,
+    requestType: NotificationSubscriptionRequestType
```

Comment:
this `requestType `parameter looks like a code smell. why do we need to disciminate the request type ? cannot we just have two different methods (the caller will discriminate)

Reply:
It exposes 2 different services that converge into the same api_request. It's not necessary to define 2 different methods.

Reply:
I agree with @gunzip, a better design would be to expose two different methods from the controller, each one implementing a specific query type, then refer to those methods from here